### PR TITLE
Scm package updates

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
Minimizing calls to GitHub, updating `fake` provider, removing and rewriting outdated methods and structs.

### CODE REVIEW

File: assignments/assignments_test.go [hunk 1]
```diff
@@ -36,17 +36,7 @@ func TestFetchAssignments(t *testing.T) {
...
-		if err != nil {
-			t.Fatal(err)
-		}
-		for _, org := range orgs {
-			if org.Path == gitHubTestOrg {
-				courseOrgID = org.ID
-				t.Logf("To speed up test; update const to 'gitHubTestOrgID = %v'", org.ID)
-			}
-		}
+		t.Fatal("Organozation ID not provided.")
```
Comment (meling) [inline assignments/assignments_test.go]: Spelling

File: scm/fake.go [hunk 2]
```diff
@@ -137,26 +129,43 @@ func (s *FakeSCM) CreateOrgHook(ctx context.Context, opt *OrgHookOptions) error
 
 // CreateTeam implements the SCM interface.
 func (s *FakeSCM) CreateTeam(ctx context.Context, opt *TeamOptions) (*Team, error) {
-	// TODO no implementation provided yet
-	return &Team{ID: 1, Name: opt.TeamName, URL: ""}, nil
+	id := len(s.Teams) + 1
```
Comment (meling) [inline scm/fake.go]: id is only used once below. move code inside the uint64() conversion directly.

File: scm/github.go [hunk 3]
```diff
@@ -31,32 +31,6 @@ func NewGithubSCMClient(logger *zap.SugaredLogger, token string) *GithubSCM {
 	}
 }
 
-// ListOrganizations implements the SCM interface.
-func (s *GithubSCM) ListOrganizations(ctx context.Context) ([]*pb.Organization, error) {
```
Comment (meling) [inline scm/github.go]: Are you sure this is not used by the scm tool? Or by the course creation process. I didn't check.

File: assignments/assignments_test.go (hunk 1 above)
Comment (0xf8f8ff) [reply assignments/assignments_test.go]: Fixed.

File: scm/fake.go (hunk 2 above)
Comment (0xf8f8ff) [reply scm/fake.go]: Done.

File: scm/github.go (hunk 3 above)
Comment (0xf8f8ff) [reply scm/github.go]: Yes.

**Pull request merged**